dotCMS/src/main/java/com/dotmarketing/fixtask/tasks/FixTask00095DeleteOrphanRelationships.java

```diff
@@ -0,0 +1,201 @@
+                modifiedData = dc.loadResults();
+                total = modifiedData != null ? modifiedData.size() : 0;
+                FixAssetsProcessStatus.setTotal(total);
+                getModifiedData();
+                if (total > 0) {
+                    startCleanup();
+                    saveFixAuditRecord(total);
+                    returnValue.add(FixAssetsProcessStatus.getFixAssetsMap());
+                    Logger.debug(
+                            FixTask00095DeleteOrphanRelationships.class,
+                            "Ending DeleteOrphanRelationships");
+                }
+            } catch (DotHibernateException e1) {
+                Logger.debug(
+                        FixTask00095DeleteOrphanRelationships.class,
+                        "A Hibernate Exception was detected during execution of DeleteOrphanRelationships task",
+                        e1);
+                HibernateUtil.rollbackTransaction();
+                FixAssetsProcessStatus.setActual(-1);
+            } catch (Exception e2) {
+                Logger.debug(
+                        FixTask00095DeleteOrphanRelationships.class,
+                        "There was an unexpected problem during execution of DeleteOrphanRelationships task",
+                        e2);
+                HibernateUtil.rollbackTransaction();
+                FixAssetsProcessStatus.setActual(-1);
+            } finally {
+                FixAssetsProcessStatus.stopProgress();
+            }
+        }
+        return returnValue;
+    }
+
+    @Override
+    public List<Map<String, String>> getModifiedData() {
+        if (modifiedData.isEmpty()) {
+            final String fixesUriSubstring = "fixes";
+            XStream xstreamObj = new XStream(new DomDriver());
+            LocalDate date = LocalDate.now();
+            SimpleDateFormat sdf = new SimpleDateFormat("dd-MM-yyyy_HH-mm-ss");
+            String lastmoddate = sdf.format(date);
+            File writingObj = null;
+            if (!new File(ConfigUtils.getBackupPath() + File.separator
+                    + fixesUriSubstring).exists()) {
+                new File(ConfigUtils.getBackupPath() + File.separator + fixesUriSubstring)
+                        .mkdirs();
+            }
+            writingObj = new File(ConfigUtils.getBackupPath() + File.separator
+                    + fixesUriSubstring + java.io.File.separator + lastmoddate + "_"
+                    + "FixTask00095DeleteOrphanRelationships" + ".xml");
+
+            BufferedOutputStream bufferedOutObj = null;
+            try {
+                bufferedOutObj = new BufferedOutputStream(new FileOutputStream(writingObj));
+            } catch (FileNotFoundException e) {
+                Logger.error(FixTask00095DeleteOrphanRelationships.class, 
+                        "Could not write to Fix Task status file.", e);
+            }
+            xstreamObj.toXML(modifiedData, bufferedOutObj);
+        }
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Return a copy of "modifiedData". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2384)